packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -193,13 +193,18 @@ export const useField = <
       return;
     }
 
-    const valueStr = event.target.value;
+    const targetValue = event.target.value;
+    const eventData = (event.nativeEvent as InputEvent).data;
+    // Calling `.fill(04/11/2022)` in playwright will trigger a change event with the requested content to insert in `event.nativeEvent.data`
+    // usual changes have only the currently typed character in the `event.nativeEvent.data`
+    const useEventData = eventData && eventData.length > 1;
```

Comment:
Are we sure it can not be more than one character? I'm thinking about special keyboards or some auto-completion

I don't see issues, it's just to share a point of attention. I first thought about #9465 but it's not the same problem since IME keyboard calls multiple times the `onChange`

Reply:
> Are we sure it can not be more than one character? I'm thinking about special keyboards or some auto-completion

I'm definitely not sure that there wouldn't be any edge cases, but autocompletion is turned off, although, if any extension hacks it's way into this, I'd imagine that this change could only fix it and not break because it would probably use a similar API of emitting a change event with the full value string. 🤔